### PR TITLE
Keep `#[CoversClass]` attribute on method, and don't move it to class scope

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/7873

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
This looks good to me 👍🏻

File: src/Rector/Class_/CoversAnnotationWithValueToAttributeRector.php [hunk 1]
```diff
@@ -120,20 +121,22 @@ private function createAttributeGroup(string $annotationValue): AttributeGroup
     }
 
     /**
+     * @param Class_|ClassMethod $node
+     *
      * @return AttributeGroup[]
      */
-    private function resolveCoversClassAttributeGroups(Class_ $class): array
+    private function resolveCoversClassAttributeGroups(Node\Stmt $node): array
```
Comment (samsonasik) [inline src/Rector/Class_/CoversAnnotationWithValueToAttributeRector.php:128]:
```suggestion
    private function resolveCoversClassAttributeGroups(Class_|ClassMethod $node): array
```

File: src/Rector/Class_/CoversAnnotationWithValueToAttributeRector.php [hunk 2]
```diff
@@ -143,39 +146,38 @@ private function resolveCoversClassAttributeGroups(Class_ $class): array
...
         return $attributeGroups;
     }
 
     /**
+     * @param Class_|ClassMethod $node
+     *
      * @return AttributeGroup[]
      */
-    private function resolveCoversFunctionAttributeGroups(Class_ $class): array
+    private function resolveCoversFunctionAttributeGroups(Node\Stmt $node): array
```
Comment (samsonasik) [inline src/Rector/Class_/CoversAnnotationWithValueToAttributeRector.php:160]:
```suggestion
    private function resolveCoversFunctionAttributeGroups(Class_|ClassMethod $node): array
```

**Review by samsonasik — APPROVED** [review]

**Review by samsonasik — CHANGES_REQUESTED** [review]
`resolveCoversClassAttributeGroups()` and `resolveCoversFunctionAttributeGroups()` has equal content, so one of them can be removed, as already cover both `Class_` and `ClassMethod`

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @staabm

**Pull request merged**